app/models/user_session.rb

```diff
@@ -1,6 +1,15 @@
 class UserSession < ApplicationRecord
 
-  TTL = 2.weeks.freeze
+  TTL = case Rails.configuration.three_scale.user_sesion_ttl
+        when nil, ""
+          2.weeks
+        when Numeric, /^\d+$/
+          # session duration specified in seconds
+          ActiveSupport::Duration.parse("PT#{Rails.configuration.three_scale.user_sesion_ttl}S")
+        else
+          # treat as ISO 8601 duration
+          ActiveSupport::Duration.parse(Rails.configuration.three_scale.user_sesion_ttl)
+        end.freeze
```

Comment:
With a a bit of DRY maybe:

```suggestion
  user_sesion_ttl = Rails.configuration.three_scale.user_sesion_ttl.presence
  TTL = begin
          case user_sesion_ttl
          when Numeric, /^\d+$/ # session duration specified in seconds
            ActiveSupport::Duration.parse("PT#{user_sesion_ttl}S")
          else
            ActiveSupport::Duration.parse(user_sesion_ttl)
          end
        rescue ActiveSupport::Duration::ISO8601Parser::ParsingError, TypeError
          # do we care about logging?
          2.weeks
        end.freeze
```

Reply:
I usually prefer to have things fail instead of silently ignoring user specified configuration. I'm also going to remove ISO8601 parsing per other comments. Do you have a strong preference to apply default configuration in case of a bad user-supplied value?

Reply:
This is not a critical feature, therefore, in case of bad user input, I'd be happy enough with falling back to a safe default plus a log message. However, this is no strong preference of mine, no. You should follow your heart 🙂

Reply:
because there are security concerns about session ttl, I'd rather fail. Please check what kind of exception I'm returning, I was not sure I chose the correct one.